geomstats/_backend/pytorch/__init__.py

```diff
@@ -296,8 +296,15 @@ def einsum(*args, **kwargs):
         if len(input_str_list) > 2:
             raise NotImplementedError(
                 'Ellipsis support not implemented for >2 input tensors')
+        ndims = [len(input_str[3:]) for input_str in input_str_list]
+
         tensor_a = input_tensors_list[0]
         tensor_b = input_tensors_list[1]
+        initial_ndim_a = tensor_a.ndim
+        initial_ndim_b = tensor_b.ndim
+        tensor_a = to_ndarray(tensor_a, to_ndim=ndims[0] + 1)
```

Comment:
what about calling einsum with the input_string without the ellipses to avoid using to_ndarray then squeeze if n_tensor = 1 ?

Reply:
Done, in PR #568.